errors.go

```diff
@@ -56,11 +56,15 @@ func SetLogger(logger Logger) error {
 
 // MySQLError is an error type which represents a single MySQL error
 type MySQLError struct {
-	Number  uint16
-	Message string
+	Number   uint16
+	SQLState string
```

Comment:
@shogo82148 How do you think making `SQLState [5]byte`?
Pros: It can avoid extra allocation.
Cons: Need cast when comparing: `string(err.SQLState) == "45000"`.

Reply:
It sounds good 👍🏼 
Most users do not need SQLState.

Reply:
SQLState is [5]byte now